tests/validation/tests/v3_api/test_monitoring.py

```diff
@@ -459,28 +465,49 @@ def create_project_client(request):
     system_project = rancher_client.list_project(clusterId=cluster.id,
                                                  name="System").data[0]
     sys_proj_client = get_project_client_for_token(system_project, USER_TOKEN)
+    cluster_client = get_cluster_client_for_token(cluster, USER_TOKEN)
     namespace["cluster"] = cluster
     namespace["project"] = project
     namespace["system_project"] = system_project
     namespace["system_project_client"] = sys_proj_client
+    namespace["cluster_client"] = cluster_client
+
+    # Deploy Longhorn app from the library catalog
+    app_name = "longhorn"
+    ns = create_ns(cluster_client, cluster, project, "longhorn-system")
+    app_ext_id = create_catalog_external_id('library', app_name,
+                                            LONGHORN_APP_VERSION)
+    answer = get_defaut_question_answers(rancher_client, app_ext_id)
+    project_client = get_project_client_for_token(project, USER_TOKEN)
+    try:
+        app = project_client.create_app(
+            externalId=app_ext_id,
+            targetNamespace=ns.name,
+            projectId=ns.projectId,
+            answers=answer)
+        print(app)
+        validate_catalog_app(project_client, app, app_ext_id, answer)
+    except (AssertionError, RuntimeError):
+        assert False, "App {} deployment/Validation failed.".format(app_name)
 
     monitoring_template = rancher_client.list_template(
         id=MONITORING_TEMPLATE_ID).data[0]
     if MONITORING_VERSION == "":
         MONITORING_VERSION = monitoring_template.defaultVersion
     print("MONITORING_VERSION=" + MONITORING_VERSION)
-    # enable the cluster monitoring
+    # Enable cluster monitoring
     if cluster["enableClusterMonitoring"] is False:
         rancher_client.action(cluster, "enableMonitoring",
                               answers=C_MONITORING_ANSWERS,
                               version=MONITORING_VERSION)
     validate_cluster_monitoring_apps()
 
-    # wait 2 minute for all graphs to be available
-    time.sleep(60 * 2)
+    # Wait 3 minutes for all graphs to be available
+    time.sleep(60 * 3)
```

Comment:
Could we poll a url or api endpoint in the deployed monitoring app (prometheus?) to make sure the graph is up instead of waiting a fixed amount of time?

Reply:
I'm ok if we need to give Rancher more time for collecting data from the monitoring service and start populating the graphs.

Reply:
The wait is to ensure there is enough data for graphs to be available.
The endpoint "URL/v3/clusters/c-cj7ng/clustermonitorgraphs.data" provides info only on the graphs but not if there is enough data to plot a graph.